src/nvim/msgpack_rpc/channel.c

```diff
@@ -321,13 +332,16 @@ static void parse_msgpack(Channel *channel)
       }
       arena_mem_free(arena_finish(&p->arena));
     } else if (p->type == kMessageTypeResponse) {
-      ChannelCallFrame *frame = kv_last(channel->rpc.call_stack);
-      if (p->request_id != frame->request_id) {
+      const char *client_type = rpc_client_type(channel);
+      ChannelCallFrame *frame = strcmp(client_type, "msgpack-rpc") == 0
```

Comment:
nit: can use strequal
```suggestion
      ChannelCallFrame *frame = strequal(client_type, "msgpack-rpc")
```